modules/dxp/apps/commerce/commerce-shop-by-diagram/commerce-shop-by-diagram-web/src/main/resources/META-INF/resources/js/Diagram.js

```diff
@@ -75,10 +77,53 @@ const Diagram = ({
 		radius: newPinSettings.defaultRadius,
 	});
 
+	const importPinSchema = (svg) => {
+		const textDatas = [];
+		const pinDatas = [];
+		const parser = new DOMParser();
+		const xmlImage = parser.parseFromString(svg, 'image/svg+xml');
+		const rootLevel = xmlImage.getElementById('Livello_Testi');
+		if (rootLevel) {
+			const rects = rootLevel.getElementsByTagName('rect');
+			const text = rootLevel.getElementsByTagName('text');
+
+			Array.from(text).map((t) => {
```

Comment:
We don't like abbreviations, can you please replace `t` with whatever it's supposed to represent. 

Also on line 96 with `r`.

Reply:
sorry you right